src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -271,6 +271,18 @@ internal void Validate()
                     "See documentation on Task Hubs for information on how to set this: https://docs.microsoft.com/azure/azure-functions/durable/durable-functions-task-hubs");
             }
 
+            string runtimeLanguage = environmentVariableResolver.Resolve("FUNCTIONS_WORKER_RUNTIME");
+            if (this.ExtendedSessionsEnabled &&
+                !string.Equals(runtimeLanguage, "dotnet", StringComparison.OrdinalIgnoreCase))
```

Comment:
GitHub doesn't always show review comment threads the way I expect, so I'll post this comment again for this latest iteration:

> Also, should we consider the case where this environment variable is not set at all? There are many .NET apps which don't set this environment variable, and I'm worried that we'd be inadvertently disabling extended sessions for those apps that upgrade to this extension version.

Reply:
I'm going to add a test for this case too.